services/repository/delete.go

```diff
@@ -133,6 +134,18 @@ func DeleteRepositoryDirectly(ctx context.Context, doer *user_model.User, repoID
 		return err
 	}
 
+	// TODO: Deleting task records could break current ephemeral runner implementation. This is a temporary workaround suggested by ChristopherHX.
+	// Since you delete potentially the only task an ephemeral act_runner has ever run, please delete the affected runners first.
+	// one of
+	//    call cleanup ephemeral runners first
+	//    delete affected ephemeral act_runners
+	//    I would make ephemeral runners fully delete directly before formally finishing the task
+	//
+	// See also: https://github.com/go-gitea/gitea/pull/34337#issuecomment-2862222788
+	if err := actions_service.CleanupEphemeralRunnersByRepoID(ctx, repoID); err != nil {
```

Comment:
Line 178, `&actions_model.ActionRunner{RepoID: repoID},` all runners will be deleted

Reply:
Yes, I mean this addition lines looks duplicated?